storage/storage_src.go

```diff
@@ -309,7 +309,7 @@ func (s *storageImageSource) LayerInfosForCopy(ctx context.Context, instanceDige
 			return nil, fmt.Errorf("reading layer %q in image %q: %w", layerID, s.image.ID, err)
 		}
 		if layer.UncompressedSize < 0 {
-			return nil, fmt.Errorf("uncompressed size for layer %q is unknown", layerID)
+			layer.UncompressedSize = -1
```

Comment:
The c/storage API does not document that the returned `layer` object can be modified by consumers. (In practice, c/storage makes a copy, so it works fine.)

Reply:
Should be fixed in #2428 .